### PR TITLE
undocker: 1.0.4 -> 1.2.2

### PR DESCRIPTION
**Pull request by motiejus**
## Description of changes

* bump v1.0.4 -> v1.2.2
* change the origin website
* add myself to the maintainers (I am the upstream author).

Co-Authored-By: John Chadwick <johnwchadwick@gmail.com>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jchv — APPROVED** [review]
LGTM.
- Just as a sanity check, I verified that the repo move seems legit. [The upstream](https://sr.ht/~motiejus/undocker/) links to the new location, and the PR author is the maintainer. So that looks good.
- Tested under x86_64-linux by using `podman image save > redis-all.tar docker.io/redis` followed by `nix run .#undocker -- ./redis-all.tar ./redis.tar`. The resulting tarball looks good.
- The diff itself LGTM from a Nix standpoint.

**Comment by motiejus** [comment]
> I think either of these approaches would be fine. It's not strictly necessary IMO, though.

Thank you, this is awesome. I copy-pasted your derivation and just put it here with a slight adjustment -- I added an explicit `make undocker` Makefile target.

I remain curious on why it needs `VSN=<...>` and why it can't pick it up from git, just like it does with sha1. But that's for another time.

**Review by jchv — APPROVED** [review]
LGTM! Now with correct version information :)

**Comment by motiejus** [comment]
I'll leave it at that. Thanks!

File: pkgs/tools/misc/undocker/default.nix [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
+    leaveDotGit = true;
   };
+in
+buildGoModule {
+  pname = "undocker";
+  inherit version src;
+
+  nativeBuildInputs = [ git gnumake ];
+
+  buildPhase = "make VSN=v${version} undocker";
```
Comment (Aleksanaa) [inline pkgs/tools/misc/undocker/default.nix:24]: So we don't need `leaveDotGit` and `git` in `nativeBuildInputs` anymore?

Comment (Aleksanaa) [reply pkgs/tools/misc/undocker/default.nix:24]: Is it that important? Can we just provide a fake hash? `leaveDotGit` frequently has reproducibility issues.

Comment (motiejus) [reply pkgs/tools/misc/undocker/default.nix:24]: Updated.

**Comment by motiejus** [comment]
I removed all git references and will use the sha256 in place of `VSNHASH`.

**Comment by Aleksanaa** [comment]
What does that VSNHASH mean? Git hash? This sha256 is a hash in SRI format, it may not be suitable here.

**Comment by motiejus** [comment]
I added VSNHASH as a debugging aide to understand where the binary is built from, it is visible in `undocker --version`. It defaults to git's sha1 hash, but, when we don't have it, we use what we have.

**Comment by Aleksanaa** [comment]
You're more likely to have problems by stuffing a value in there that clearly doesn't belong there. We already have a VSN, so let’s put a hash there that is obviously wrong, for example, all 0s.

**Review by jordanisaacs — APPROVED** [review]

**Pull request merged**